### PR TITLE
Optimize latest project update query

### PR DESCRIPTION
**Pull request by CarlosQ96**
Related to:https: //github.com/Giveth/impact-graph/issues/1086

### CODE REVIEW

File: src/resolvers/projectResolver.ts [hunk 1]
```diff
@@ -1799,14 +1799,14 @@ export class ProjectResolver {
...
-        LIMIT 1
-      )
+      FROM public.project_update AS pu
+      JOIN (
+          SELECT "projectId", MAX("createdAt") AS max_createdAt
+          FROM public.project_update
+          WHERE "isMain" = false
+          GROUP BY "projectId"
+      ) AS puu_max
+      ON pu."projectId" = puu_max."projectId" AND pu."createdAt" = puu_max.max_createdAt
```
Comment (aminlatifi) [inline src/resolvers/projectResolver.ts]: As I see the Join is not necessary! No further information is added outside and both tables are same `project_update`.
@CarlosQ96 Do we have tests for this? can you add if there isn't and then test without join statement?
Something like this
```sql
 SELECT "projectId", MAX("createdAt") AS max_createdAt
          FROM public.project_update
          WHERE "isMain" = false
          GROUP BY "projectId"
```

**Review by aminlatifi — CHANGES_REQUESTED** [review]

**Comment by CarlosQ96** [comment]
Yes Amin there are tests for this. However, I think one can be improved a lot so ill finish the test and optimize more the query.

**Comment by MoeNick** [comment]
Hey @CarlosQ96 
Amin already requested some change to this. 
Did  you review it?

**Comment by CarlosQ96** [comment]
Further optimization from what I did in this PR, breaks a bunch of tests. I am somewhat stuck on that fixing tests.

File: src/resolvers/projectResolver.ts (hunk 1 above)
Comment (CarlosQ96) [reply src/resolvers/projectResolver.ts]: @aminlatifi I implemented your suggestion, modified it a little and it works fine.

I also tested my join version (previous) and it worked fine.
The tests are in the projectResolver.test.ts (those worked with the original implementation too)

Can review again.

**Comment by CarlosQ96** [comment]
@MoeNick @aminlatifi can review again

File: src/resolvers/projectResolver.ts [hunk 2]
```diff
@@ -1799,17 +1799,12 @@ export class ProjectResolver {
...
-        SELECT puu.id
-        FROM project_update as puu
-        WHERE puu."isMain" = false AND pu."projectId" = puu."projectId"
-        ORDER BY puu."createdAt" DESC
-        LIMIT 1
-      )
-      ORDER BY pu."createdAt" DESC
+      FROM public.project_update AS pu
+      WHERE pu."isMain" = false
+      GROUP BY pu.id, pu."projectId"
```
Comment (aminlatifi) [inline src/resolvers/projectResolver.ts]: Are you sure it must be grouped by both `id` and `projectId`?! maybe just `projectId`

**Review by aminlatifi — CHANGES_REQUESTED** [review]

File: src/resolvers/projectResolver.ts (hunk 2 above)
Comment (CarlosQ96) [reply src/resolvers/projectResolver.ts]: You are right Amin I removed the double grouping. I only need the projectUpdate.projectId

**Review by aminlatifi — APPROVED** [review]
LGTM

**Pull request merged**